flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/CalcITCase.scala

```diff
@@ -1837,6 +1837,11 @@ class CalcITCase extends BatchTestBase {
       Seq(row('a'), row('b'), row('a')))
   }
 
+  @Test
+  def testIfWithFixCharAndVarchar(): Unit = {
+    checkResult("SELECT IF(b > 10, 'ua', c) FROM Table3", data3.map(r => row(r.getField(2))))
```

Comment:
Change to 
```suggestion
    checkResult("SELECT IF(b > 4, 'ua', c) FROM Table3", data3.map(r => row(r.getField(2))))
```
to cover the negative case.